mitodl/open-discussions
# fix ci commands; remove flowgen

#### Pre-Flight checklist


#### What are the relevant tickets?
Fixes (🤞) deploy error

#### What's this PR do?
This PR:
- fixes `yarn-run-with-fallback.js` to exit with error code 1 if there's an error. Previously it was always exiting with code 0 :scream:
- Removes `flowgen`. I *believe* this was the cause of the build error, but I am not 100% sure.

    Flowgen (https://github.com/joarwilk/flowgen) is a tool for converting ts types to flow types. Without something like this, whenever we extract js code from `open-discussions` into ts for sharing with infinite-corridor, its flow type will be `any`. My hope was that Flowgen would let us use Typescript for the shared code but generate flow types for backwards compatibility in open-discussions.

    The trouble is with imported types, like Typescript's `React.ChangeEventHandler`, or `React.ReactNode`, or `React.FC<Props>`. Flowgen does OK with some of these, but it fails to convert others, which resulted in Flow errors like
    ```
    // When flowgen fails to convert a typescript type, we got a flow error
    Cannot get `React.ReactNode` because property `ReactNode` is missing in object type [1]
    ```

    Here's the error on a previous CI run, but it did not cause the test to fail because of the issue above. https://github.com/mitodl/open-discussions/runs/7217323166?check_suite_focus=true#step:11:35

    Anyway... For now I think we should just remove Flowgen. A pretty high number of the flow types in `opne-discussions` seem to be `any` or very loose anyway.

#### How should this be manually tested?
1. Cause some formatting or linting error... e.g., add a semicolon or an unused import. And run `docker compose run --rm watch yarn run fmt-check` (or `lint-check`). The process should exit with code 1.
2. Also... [More coming]
